lib/puma/launcher.rb

```diff
@@ -59,7 +59,7 @@ def initialize(conf, launcher_args={})
 
       @environment = conf.environment
 
-      if ENV["NOTIFY_SOCKET"]
+      if ENV["NOTIFY_SOCKET"] && RUBY_PLATFORM != "java"
```

Comment:
Makes sense seeing the tests are skipped on JRuby: https://github.com/puma/puma/blob/e3d5794a7ebe47577ced4d4dfdd6a6cc969ded01/test/test_plugin_systemd.rb#L13

I do wonder though, can we make the skip more specific? 

Similar to `Puma::HAS_UNIX_SOCKET` / `skip_unless :unix`: https://github.com/puma/puma/blob/e3d5794a7ebe47577ced4d4dfdd6a6cc969ded01/lib/puma.rb#L29

(and also skip the tests based on this?)

Reply:
@mohamedhafez Can we at least add a comment why we exclude JRuby here?

Should probably also make use of the detection Puma already has?

```suggestion
      if ENV["NOTIFY_SOCKET"] && Puma.jruby?
```

Reply:
Didn't know about `Puma.jruby?`, added that in (but it needs to be `!Puma.jruby?`). Also good call with the comment, so people know why this was needed in the future, and also I put the issue in the comment so that if anyone in the future is considering removing it they can double check if the issue was resolved easily.